storage/client/blob/src/samples/java/blob/ListContainersExample.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package blob;
+
+import com.azure.core.util.configuration.ConfigurationManager;
+import com.azure.storage.blob.StorageClient;
+import com.azure.storage.common.credentials.SharedKeyCredential;
+import java.util.Locale;
+
+/**
+ * This example shows how to list all containers with storage client using the Azure Storage Blob SDK for Java.
+ */
+public class ListContainersExample {
+    public static void main (String[] args) {
+        String accountName = getAccountName();
+        String accountKey = getAccountKey();
+
+        /**
+         * Use your Storage account's name and key to create a credential object; this is used to access your account.
+         */
+        SharedKeyCredential credential = new SharedKeyCredential(accountName, accountKey);
+
+        /**
+         * From the Azure portal, get your Storage account blob service URL endpoint.
+         * The URL typically looks like this:
+         */
+        String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
+
+        /**
+         * Create a StorageClient object that wraps the service endpoint, credential and a request pipeline.
+         */
+        StorageClient storageClient = StorageClient.storageClientBuilder().endpoint(endpoint).credential(credential).buildClient();
+
+        /**
+         * Create 3 different containers from the storageClient.
+         */
+        for (int i = 0; i < 3; i++) {
+            storageClient.createContainer("mycontainersforlisting" + i + System.currentTimeMillis());
+        }
+
+        /**
+         * List the containers' name under the Azure storage account.
+         */
+        storageClient.listContainers().forEach(
+            containerItem ->
+            {
+                System.out.println("Container name: " + containerItem.name());
+
+                /**
+                 * Clean up the containers at the same time.
+                 */
+                storageClient.getContainerClient(containerItem.name()).delete();
```

Comment:
Should we show off StorageClient.deleteContainer here instead of ContainerClient.delete?

Reply:
Especially if we already are at the storage client instead of the container client. We should, however, leave a comment saying that it's a shortcut for the code currently here.

Reply:
Done. Removed for now.

Reply:
I have tried to delete at the storage client level. I did not find the deleteContainer API. Are we supposed to have the API?

Reply:
Huh, it does appear to be missing deleteContainer, open up an issue for that and we'll resolve.

@jaschrep-msft and @rickle-msft good with resolving this in preview 2?